File: 26th-April-Seating_Arrangement/c++/solution.cpp
/*
  This code is written by Bhaskar
*/


//User function Template for C++
class Solution{
    public:
    bool is_possible_to_get_seats(int n, int m, vector<int>& seats){
        // Write your code here.
        int cnt = 0;
        
    for (int i = 0; i < m; i++) {
        
        if(i==0){
            if(seats[i]+seats[i+1]==0){
                cnt++;
                seats[i]=1;
            }
        }else if(i==m-1){
            if(seats[i]+seats[i-1]==0){
                cnt++;
                seats[i]=1;
            }
        }else{
            if(seats[i-1]+seats[i]+seats[i+1]==0){
                cnt++;
                seats[i]=1;
            }
        }
        
        if (cnt >= n) return true;
    }
    return false;
}

File: 26th-April-Seating_Arrangement/java/solution.java
/*
  This code is written by Bhaskar
*/



class Solution {
    public static boolean is_possible_to_get_seats(int n, int m, int[] seats) {
        // code here
       int cnt = 0;
        for (int i = 0; i<m; i++) {
            int pre;
            if(i==0) pre=0;
            else pre=seats[i-1];

            int next;
            if(i==m-1) next=0;
            else next=seats[i+1];
            
            if (pre+next+seats[i]==0) {
                cnt++;
                i++;
            } 
        }
        return cnt>=n ? true : false;
    }
}                         
            
            
